embedded-graphics/src/image/mod.rs

```diff
@@ -3,31 +3,151 @@
 //! Image drawables can be created for raw bitmap data and images in BMP and TGA
 //! format.
 
-#[cfg(feature = "bmp")]
-mod image_bmp;
+// #[cfg(feature = "bmp")]
+// mod image_bmp;
 mod image_raw;
-#[cfg(feature = "tga")]
-mod image_tga;
+// #[cfg(feature = "tga")]
+// mod image_tga;
 
-pub use self::image_raw::{Image, ImageBE, ImageLE};
+pub use self::image_raw::{ImageBE, ImageLE, ImageRaw};
 
-#[cfg(feature = "bmp")]
-pub use self::image_bmp::ImageBmp;
-#[cfg(feature = "tga")]
-pub use self::image_tga::ImageTga;
+use crate::draw_target::DrawTarget;
+use crate::drawable::Drawable;
+use crate::drawable::Pixel;
+use crate::geometry::Dimensions;
+use crate::geometry::Point;
+use crate::geometry::Size;
+use crate::pixelcolor::PixelColor;
+use crate::transform::Transform;
+use core::marker::PhantomData;
 
-/// Image file trait.
-pub trait ImageFile<'a>: crate::geometry::Dimensions + Sized {
-    /// Create a new image with given input file
-    ///
-    /// The input file is expected to be of a particular format (BMP, TGA, etc) and contain file
-    /// metadata like width/height and pixel data. Because parsing may fail, this returns a
-    /// `Result<Self, ()>`.
-    fn new(filedata: &'a [u8]) -> Result<Self, ()>;
+/// TODO: Docs
+pub trait ImageData<C>
+where
+    C: PixelColor + From<<C as PixelColor>::Raw>,
+{
+    /// Iterator over pixels in the image
+    type PixelIterator: Iterator<Item = Pixel<C>>;
 
     /// Get the width in pixels of an image
     fn width(&self) -> u32;
 
     /// Get the height in pixels of an image
     fn height(&self) -> u32;
+
+    /// Get an iterator over the pixels of the image
+    fn pixel_iter(&self) -> Self::PixelIterator;
+}
+
+/// TODO: Docs
+#[derive(Debug, Clone, Copy)]
+pub struct Image<I: Clone, C> {
```

Comment:
The `Clone` requirement is required for `Transform::translate_mut`. This can be fixed and removed in #211, but let's do it after this PR is merged.

Reply:
Also, I renamed `Image` in `image_raw.rs` to `ImageRaw` and reused the `Image` name here for the wrapping type. Now that I write this, this will make migration a pain so do we leave it, or put it in the changelog and just stomach it? I'm erring on the latter seeing as this release is full of so many breaking changes already.